src/journal/bsod.c

```diff
@@ -75,13 +78,23 @@ static int acquire_first_emergency_log_message(char **ret) {
         if (r < 0)
                 return log_error_errno(r, "Failed to seek to start of jornal: %m");
 
-        r = sd_journal_next(j);
-        if (r < 0)
-                return log_error_errno(r, "Failed to read next journal entry: %m");
-        if (r == 0) {
-                log_debug("No emergency level entries in the journal");
-                *ret = NULL;
-                return 0;
+        for(;;) {
+                r = sd_journal_next(j);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to read next journal entry: %m");
+
+                if (r == 0) {
```

Comment:
in the interesting of keeping nesting levels shallow please invert this check and the next, i.e.:

```c
r = sd_journal_next(j);
if (r < 0)
        return …
if (r > 0)
        break;
if (!arg_continuous) {
        log_debug(…);
        *ret = NULL;
        return 0;
}

r = sd_journal_wait(…
```

you get the idea

Reply:
Yes sir. Super cool!!!